src/bls12_381/ec.rs

```diff
@@ -1557,20 +1557,70 @@ pub mod g2 {
             }
         }
 
-        fn scale_by_cofactor(&self) -> G2 {
-            // G2 cofactor = (x^8 - 4 x^7 + 5 x^6) - (4 x^4 + 6 x^3 - 4 x^2 - 4 x + 13) // 9
-            // 0x5d543a95414e7f1091d50792876a202cd91de4547085abaa68a205b2e5a7ddfa628f1cb4d9e82ef21537e293a6691ae1616ec6e786f0c70cf1c38e31c7238e5
-            let cofactor = BitIterator::new([
-                0xcf1c38e31c7238e5,
-                0x1616ec6e786f0c70,
-                0x21537e293a6691ae,
-                0xa628f1cb4d9e82ef,
-                0xa68a205b2e5a7ddf,
-                0xcd91de4547085aba,
-                0x91d50792876a202,
-                0x5d543a95414e7f1,
+
```

Comment:
Add a comment referencing the paper this comes from, and briefly explaining what's going on.